src/shared/varlink-io.systemd.Metrics.c

```diff
@@ -0,0 +1,64 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "sd-varlink-idl.h"
+
+#include "varlink-io.systemd.Metrics.h"
+
+/* TODO Let's use this as a temporary solution */
+static SD_VARLINK_DEFINE_STRUCT_TYPE(
+                MetricFields,
+                SD_VARLINK_FIELD_COMMENT("State of the unit. Applicable for X metrics"),
+                SD_VARLINK_DEFINE_FIELD(state, SD_VARLINK_STRING, SD_VARLINK_NULLABLE),
+                SD_VARLINK_DEFINE_FIELD(load_state, SD_VARLINK_STRING, SD_VARLINK_NULLABLE),
+                SD_VARLINK_DEFINE_FIELD(sub_state, SD_VARLINK_STRING, SD_VARLINK_NULLABLE),
+                SD_VARLINK_DEFINE_FIELD(freezer_state, SD_VARLINK_STRING, SD_VARLINK_NULLABLE),
+                SD_VARLINK_DEFINE_FIELD(unit_file_state, SD_VARLINK_STRING, SD_VARLINK_NULLABLE));
+
+static SD_VARLINK_DEFINE_STRUCT_TYPE(
+                Metric,
+                SD_VARLINK_FIELD_COMMENT("Metric name"),
+                SD_VARLINK_DEFINE_FIELD(name, SD_VARLINK_STRING, 0),
+                SD_VARLINK_FIELD_COMMENT("Metric value"),
+                SD_VARLINK_DEFINE_FIELD(value, SD_VARLINK_INT, 0),
+                SD_VARLINK_FIELD_COMMENT("Object name"),
+                SD_VARLINK_DEFINE_FIELD(object, SD_VARLINK_STRING, 0),
+                SD_VARLINK_FIELD_COMMENT("Metric fields"),
+                SD_VARLINK_DEFINE_OUTPUT_BY_TYPE(fields, MetricFields, SD_VARLINK_NULLABLE));
+
+static SD_VARLINK_DEFINE_STRUCT_TYPE(
+                MetricMetadata,
+                SD_VARLINK_FIELD_COMMENT("Metric name"),
+                SD_VARLINK_DEFINE_FIELD(name, SD_VARLINK_STRING, 0),
+                SD_VARLINK_FIELD_COMMENT("Metric description"),
+                SD_VARLINK_DEFINE_FIELD(value, SD_VARLINK_STRING, 0),
+                SD_VARLINK_FIELD_COMMENT("Metric type"),
+                SD_VARLINK_DEFINE_FIELD(type, SD_VARLINK_STRING, 0));
+
+static SD_VARLINK_DEFINE_ERROR(NoSuchMetric);
+
+static SD_VARLINK_DEFINE_METHOD_FULL(
+                List,
+                SD_VARLINK_SUPPORTS_MORE,
+                SD_VARLINK_DEFINE_OUTPUT_BY_TYPE(metrics, Metric, 0));
```

Comment:
There is no need to 'metrics' field here (and hence separate Metric) type. Check how structured `List` methods in `src/shared/varlink-io.systemd.Machine.c` or `src/shared/varlink-io.systemd.Unit.c`